### PR TITLE
Health Tracking System

### PR DESCRIPTION
**Pull request by Thiru5**
- added list of heart images
- added tracking of health points by list of heart images
- added healthcontroller.cs which increments and decrements the list of heart images
- adjusted minor bug in health.cs for respawn health

### CODE REVIEW

File: Assets/Scripts/Mechanics/HealthController.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            hearts[0].enabled = true;
+            hearts[1].enabled = true;
+            hearts[2].enabled = true;
+            livesRemaining = livesRemaining + 3;
+        }
+    }
+
+
+    public void Update(){
+        if(Input.GetKeyDown(KeyCode.Return)){
```
Comment (seaerchin) [inline Assets/Scripts/Mechanics/HealthController.cs]: this bugs on keydown of enter etc - so when a player is walking through the map, if this happens, the health will decrement

File: Assets/Scripts/Mechanics/Health.cs [hunk 2]
```diff
@@ -5,11 +5,15 @@
...
 namespace Platformer.Mechanics
 {
+    
     /// <summary>
     /// Represebts the current vital statistics of some game entity.
     /// </summary>
     public class Health : MonoBehaviour
     {
+
+        public PlayerController player;
```
Comment (seaerchin) [inline Assets/Scripts/Mechanics/Health.cs]: i don't think this circular reference is needed - player can already obtain a ref through using ```GetComponent```

File: Assets/Scripts/Gameplay/PlayerEnemyCollision.cs [hunk 3]
```diff
@@ -15,6 +15,7 @@ public class PlayerEnemyCollision : Simulation.Event<PlayerEnemyCollision>
     {
         public EnemyController enemy;
         public PlayerController player;
+        public HealthController healthCon;
```
Comment (seaerchin) [inline Assets/Scripts/Gameplay/PlayerEnemyCollision.cs]: no point having this here cos it's done below and never persisted

File: Assets/Scripts/Mechanics/ScoreController.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    // Update is called once per frame
+    // just poll and update
+    void Update()
+    {
+        token_score.text = token.getCount.ToString();
+        enemy_score.text = _enemy_score.ToString();
+        Debug.Log("Enemy" + _enemy_score);
+    }
+}
```
Comment (seaerchin) [inline Assets/Scripts/Mechanics/ScoreController.cs]: isn't this already in dev?

File: Assets/Scripts/Mechanics/HealthController.cs (hunk 1 above)
Comment (Thiru5) [reply Assets/Scripts/Mechanics/HealthController.cs]: removed it in the next update

File: Assets/Scripts/Mechanics/Health.cs (hunk 2 above)
Comment (Thiru5) [reply Assets/Scripts/Mechanics/Health.cs]: ok rectified

File: Assets/Scripts/Mechanics/ScoreController.cs (hunk 4 above)
Comment (Thiru5) [reply Assets/Scripts/Mechanics/ScoreController.cs]: yea im not sure why it was tagged as an update LOL

File: Assets/Scripts/Gameplay/PlayerEnemyCollision.cs (hunk 3 above)
Comment (Thiru5) [reply Assets/Scripts/Gameplay/PlayerEnemyCollision.cs]: ok rectifed

**Pull request merged**